### ISSUE TITLE
quickstart: cant remove and recreate the index, whilst its serving.

### ISSUE THREAD

**Issue by gedw99** [quickwit-inc/quickwit#820]
**Describe the bug**

Cant delete indexes while serve is running.

**Steps to reproduce (if applicable)**

1. Follow the https://quickwit.io/docs/getting-started/quickstart/
2. Can do curl and rest based searches with no issues. 
3. Delete the index.  
"quickwit delete --index-uri file:///$(DATA_FSPATH)/wikipedia"

4. Create the index. 
mkdir -p $(DATA_FSPATH)
quickwit new --index-uri file:///$(DATA_FSPATH)/wikipedia --index-config-path $(EX_FSPATH)/wikipedia_index_config.json
quickwit index --index-uri file:///$(DATA_FSPATH)/wikipedia --input-path $(EX_FSPATH)/wiki-articles-10000.json

5. Run a Curl or rest query.
curl "http://0.0.0.0:8080/api/v1/wikipedia/search?query=barack+AND+obama"

**Expected behavior**

I would have thought removing indexes via the CLI, and then adding them whilst the server is running would be ok.
If i stop the server and restart it, the query works.

**System configuration:**
Quickwit 0.1.0
Mac intel.

**Additional context**

**Comment by gedw99** [comment]
@fmassot thanks. 

when using file based storage, the server is not aware when the index is delete.
so need to use object store instead.


Maybe my use case of using file based store will be supported . Let me know if this is going to be supported.

**Comment by gedw99** [comment]
I attach a makefile that reproduces it.

See failure point....


makefile:
``` 

BIN_FSPATH=$(PWD)/bin

QW_BIN_NAME=quickwit
QW_BIN_FSPATH=$(BIN_FSPATH)/qw
QW_BIN=$(QW_BIN_FSPATH)/$(QW_BIN_NAME)




print:
	@echo
	@echo QW_BIN : $(QW_BIN)

dep:
	curl -L https://install.quickwit.io | sh
	mkdir -p $(QW_BIN_FSPATH)
	cp ./quickwit $(QW_BIN_FSPATH)
	rm ./quickwit
	
run-version:
	$(QW_BIN) --version

run-serve:
	$(QW_BIN) serve


### EXAMPLE Data

EX_FSPATH=$(PWD)/examples

ex-delete:
	rm -rf EX_FSPATH

ex-dep:
	# First, download the Wikipedia config from Quickwit repository.
	mkdir -p $(EX_FSPATH)
	curl -o $(EX_FSPATH)/wikipedia_index_config.json https://raw.githubusercontent.com/quickwit-inc/quickwit/main/examples/index_configs/wikipedia_index_config.json

	# add some docs
	# Download the first 10_000 Wikipedia articles.
	curl -o $(EX_FSPATH)/wiki-articles-10000.json https://quickwit-datasets-public.s3.amazonaws.com/wiki-articles-10000.json

### DATA

DATA_FSPATH=$(PWD)/data

data-delete:
	rm -rf $(DATA_FSPATH)

data-index-create:
	# create an index to the config.
	mkdir -p $(DATA_FSPATH)
	$(QW_BIN) new --index-uri file:///$(DATA_FSPATH)/wikipedia --index-config-path $(EX_FSPATH)/wikipedia_index_config.json
	
	# Index our 10k documents.
	$(QW_BIN) index --index-uri file:///$(DATA_FSPATH)/wikipedia --input-path $(EX_FSPATH)/wiki-articles-10000.json

data-index-delete:
	$(QW_BIN) delete --index-uri file:///$(DATA_FSPATH)/wikipedia

data-query:
	$(QW_BIN) search --index-uri file:///$(DATA_FSPATH)/wikipedia --query "barack AND obama"

data-serve:
	$(QW_BIN) serve --index-uri file:///$(DATA_FSPATH)/wikipedia
	# http://0.0.0.0:8080/api/v1/wikipedia/search

server-query:
	curl "http://0.0.0.0:8080/api/v1/wikipedia/search?query=barack+AND+obama"

	curl "http://0.0.0.0:8080/api/v1/wikipedia/search?query=body:barack+AND+obama"
``` 

open 2 terminals

terminal 1:
````

make dep

make run-version

make ex-dep

make data-delete

make data-index-create

make data-query
# works :)

make data-serve
# orks :)

````


terminal 2:
````

make server-query
# works :)

make data-index-delete

make data-index-create

make server-query
# failure !

# Have to start and stop server, and then the  "make server-query" works



````

**Comment by fmassot** [comment]
@gedw99 this has been fixed by #839
So we are polling at a given interval the index metadata when you are using a file backed metastore (s3 or local file system).

`s3://my_bucket/blabla#period=30s`
